docs/content/concepts/metadata-tags/asset-metadata/table-metadata.mdx

```diff
@@ -0,0 +1,192 @@
+        for name, column_type in zip(column_names, column_types)
+    ]
+
+    yield MaterializeResult(
+        metadata={"dagster/column_schema": TableSchema(columns=columns)}
+    )
+```
+
+The schema for `my_asset` will be visible in the Dagster UI. You may optionally attach <PyObject object="TableColumnConstraints"/> to each column to provide additional context about the values in the column:
+
+```python file=/concepts/metadata-tags/asset_column_schema_constraints.py
+from dagster import (
+    AssetKey,
+    MaterializeResult,
+    TableColumn,
+    TableColumnConstraints,
+    TableSchema,
+    asset,
+)
+
+
+@asset(
+    deps=[AssetKey("source_bar"), AssetKey("source_baz")],
+    metadata={
+        "dagster/column_schema": TableSchema(
+            columns=[
+                TableColumn(
+                    "name",
+                    "string",
+                    description="The name of the person",
+                ),
+                TableColumn(
+                    "age",
+                    "int",
+                    description="The age of the person",
+                    constraints=TableColumnConstraints(nullable=False, other=[">0"]),
+                ),
+            ]
+        )
+    },
+)
+def my_asset(): ...
+```
+
+### For assets loaded from integrations
+
+Column schemas are currently supported in the dbt integration. Refer to the [dbt documentation](/integrations/dbt/reference) for more information.
+
+---
+
+## Attaching row count
+
+### For assets defined in Dagster
+
+<Image
+alt="Row count for an asset in the Dagster UI"
+src="/images/concepts/metadata-tags/metadata-row-count.png"
+width={1921}
+height={559}
+/>
+
+You may attach row count schema metadata to Dagster assets as [materialization metadata](/concepts/metadata-tags/asset-metadata#attaching-materialization-metadata) to provide additional context about the number of rows in a materialized table. Dagster will let you track changes in the row count over time, and you can use this information to monitor data quality.
+
+To attach schema metadata to an asset, you will need to attach a numerical value to the `dagster/row_count` key in the metadata parameter of the <PyObject object="MaterializeResult" /> object returned by the asset function.
+
+Below is an example of how to attach row count metadata to an asset:
+
+```python file=/concepts/metadata-tags/asset_row_count.py
+import pandas as pd
+
+from dagster import AssetKey, MaterializeResult, asset
+
+
+@asset(deps=[AssetKey("source_bar"), AssetKey("source_baz")])
+def my_asset():
+    my_df: pd.DataFrame = ...
+
+    yield MaterializeResult(metadata={"dagster/row_count": 374})
+```
+
+---
+
+## Attaching column-level lineage
+
+Column lineage enables data and analytics engineers alike to understand how a column is created and used in your data platform. Refer to the [Column-level lineage documentation](/concepts/metadata-tags/asset-metadata/column-level-lineage) for more information.
```

Comment:
I think we should just merge that with this page. Column-level lineage is a kind of table metadata.

Reply:
I'm inclined to keep them separate. The page could become long and unwieldy otherwise, and having a dedicated column lineage page gives us a page-level heading (versus a section level, which will be a lower rank) to use in search results.